hs_app_timeseries/tests/views/test_create_timeseries_resource.py

```diff
@@ -0,0 +1,141 @@
+                     'irods_federated': 'true',
+                     'files': (self.odm2_sqlite_file_name, open(self.odm2_sqlite_file))
+                     }
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
+        self.assertEqual(ResourceFile.objects.count(), 1)
+        ts_resource = BaseResource.objects.filter(short_id=res_id).first()
+        # check that the resource title got updated due to metadata extraction as part of resource
+        # creation
+        self.assertEqual(ts_resource.metadata.title.value,
+                         "Water temperature data from the Little Bear River, UT")
+        hydroshare.delete_resource(res_id)
+
+    def test_create_resource_with_invalid_file(self):
+        # here we are testing the create_resource view function
+
+        self.assertEqual(BaseResource.objects.count(), 0)
+        self.assertEqual(ResourceFile.objects.count(), 0)
+        # test with bad sqlite file - this file should not be uploaded
+        post_data = {'resource-type': 'TimeSeriesResource',
+                     'title': 'Test Time Series Resource Creation',
+                     'irods_federated': 'true',
+                     'files': (self.odm2_sqlite_invalid_file_name,
+                               open(self.odm2_sqlite_invalid_file))
+                     }
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
+        # that bad sqlite file was not uploaded
+        self.assertEqual(ResourceFile.objects.count(), 0)
+        hydroshare.delete_resource(res_id)
+
+    def _set_request_message_attributes(self, request):
+        # the following 3 lines are for preventing error in unit test due to the view being
+        # tested uses messaging middleware
+        setattr(request, 'session', 'session')
+        messages = FallbackStorage(request)
+        setattr(request, '_messages', messages)
+
+    def _add_session_to_request(self, request):
+        """Annotate a request object with a session"""
+        middleware = SessionMiddleware()
+        middleware.process_request(request)
+        request.session.save()
```

Comment:
Above two internal method ```_set_request_message_attributes``` and ```_add_session_to_request``` should be consolidated into common methods that can be used by view tests of different resource types to eliminate duplications.